presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -466,6 +486,61 @@ private RelationPlan createTableWriterPlan(
         return new RelationPlan(commitNode, analysis.getRootScope(), commitNode.getOutputSymbols());
     }
 
+    /*
+    According to the standard, for the purpose of store assignment (INSERT),
+    no non-space characters of a character string, and no non-zero octets
+    of a binary string must be lost when the inserted value is truncated to
+    fit in the target column type.
+    The following method returns a cast from source type to target type
+    with a guarantee of no illegal truncation.
+    TODO Once BINARY and parametric VARBINARY types are supported, they should be handled here.
+    TODO This workaround is insufficient to handle structural types
+     */
+    private Expression noTruncationCast(Expression fromExpression, Type fromType, Type toType)
+    {
+        if (fromType instanceof UnknownType || (!(toType instanceof VarcharType) && !(toType instanceof CharType))) {
+            return new Cast(fromExpression, toSqlType(toType));
+        }
+        int targetLength;
+        if (toType instanceof VarcharType) {
+            if (((VarcharType) toType).isUnbounded()) {
+                return new Cast(fromExpression, toSqlType(toType));
+            }
+            targetLength = ((VarcharType) toType).getBoundedLength();
+        }
+        else {
+            targetLength = ((CharType) toType).getLength();
+        }
+
+        checkState(fromType instanceof VarcharType || fromType instanceof CharType, "inserting non-character value to column of character type");
+        Type regexType = regexLibrary.name().equals("JONI") ? JONI_REGEXP : RE2J_REGEXP;
```

Comment:
What's the purpose of parameterizing the regex implementation choice?

Reply:
I need it to resolve the `regex_replace` function.